##Okay, well, here is my *ATTEMPT* at coding a program. This script is for a C/G Percent Calculator, using the perl language. 

####Why write this in perl?
I originally wrote the script in Python, but I couldn't manage to get it to work quite right. After much crying and hair-pulling out trying to troubleshoot my code, I scrapped the original and decided to start over. I went online for some help and found [a code](http://www.techcuriosity.com/resources/bioinformatics/frequency_of_nucleotide.php) in Perl that did a total nucleotide count for a single file. I figured that was as good a place to start as any. If I taught myself Python in a weekend, one more language couldn't be that bad, right? I WAS VERY WRONG. At the end of the day, however, I like this version much more than my original Python code. The Perl and Python languages have basic similarities, but the main differences were that Perl relied more on symbology and Python on words and that Python programs are much more reusable. They don't need as many changes to be switched to a different task, whereas Perl programs are often one code for one task, but are highly efficient at that one task. Regardless, both languages seem to be equally useful for Bioinformatics, (esp. the BioPython and BioPerl modules), so it doesn't really seem to matter which language you prefer.

##What *is* this  C/G Percent Calculator and *why* do we even want one?
DNA (where we store our genetic material) comes in long strands of molecules called nucleotides. There are 4 kinds of nucleotides, A (Adenine), T (Thymine), G (Guanine), and C (Cytosine). A pairs up with T and C pairs up with G. 3 of these nucleotides together are called codons. A C/G Percent Calculator should be able to count the number C nucleotides, the number of G nucleotides, and the total number of nucleotides in the sequence, and use these numbers to calculate how much of a given genetic sample is Cs and Gs. Knowing the C/G percent content of a sequence can be very useful for researchers. Since Start and Stop codons, generally placed at the beginning and end of coding sequences, contain mostly As and Ts, the regions with more Cs and Gs may correspond to coding portion of genes that are of interest. Higher levels of CG content also correspond to higher melting points, if researchers are interested in DNA denaturation. 



## Alright, let's get started. 

####First we need to tell the computer we are using a perl script, not the normal bash script. 
Also, we're going to tell the computer to call up a pragma (also known as a directive) called *Strict*. This is a default mode that enables backwards compatibility with older versions of perl. It also has some functions that stop me from making complete mess of 
everything with my (inevitable) incompetence.

    #!/usr/bin/perl 
    use strict; 

####Now, we need to open up a file to analyze. 
The *ARGV* command tells program to go get whichever file is specified on the command line. *die* will stop the program if no file is provided.

    my $file = "$ARGV[0]";
    open FILE, $file or die "$file: $!\n";

####Our file has multiple entries, so we need some way to match our output values with their approriate record. 
Here, we'll start up the label making process, using headers from the entries in the file to create output labels.

    my $header;

###We have multiple sequences in our file, and we want to read each one separately. 
That means we need to create a loop. Loops allow us to create a chain of commands, and then tell the program that whenever a specified set of conditions is met, it needs to execute these commands. The alternative is going back and re-entering the commands whenever we want to use them.

####Here, we're going to set up a *while* loop. 
A *while* loop is sort of a repeating statement. "While x is true, do y." 
Our while loop is going to help us differenetiate between the sequence that we are interested in and the headers we are 
using as labels. The *chomp* command removes any trailing record separater from a string so the rest of the sequence can be read.

    while (my $line = <FILE>){ 
      chomp $line;
####Now, we want to create an array so we can store the numbers we will generate later in the program. 
The *split* command splits up the string generated by the array into a more table-like format that will be easier for the program to attach our labels
to later on.
   
      my @seq = split (/>/, $line);
    
####Before we calculate anything, we should finish creating our labels.
To do this, we're going to set an "if" loop inside our "while" loop. Loopception. An "if" loop reads as "if x is true, do y. if x is not true ("else"), do z". Since this is part of our larger "while" loop, 
  this command will check each line. If first symbol is a ">", it won't run the analysis we are about to set up. Instead, it will save 
  the data on that line as one of our labels, then move on to the next line.
    
      if ($line =~ />(.*)/){
          $header = $1

####On to the main event, calculating CG Percent.
This is the "else" portion of our "if" loop. If the line wasn't used in the first statement above, we're going to use it here. Below we have lines that calculate the number of Cs and Gs in each in each entry, adds those numbers together, counts the total number of bases, then takes the total number of Cs and Gs and divides that number by the total base number. Then it multiples that value by 100 to give us our percent.
   
      } else{
      if ($line !~ />/){
         my $G = ($line =~ tr/G//);
         my $C = ($line =~ tr/C//);
         my $total = $G + $C;
         my $length = length($line);
         my $percent = ($total / $length) * 100;

####Finally, we've got our numbers *and* our labels. 
So now we need to put them together in a manner that is actually readable to the average person. Tables are nice, but I like sentences, so that's what we'll have the program print out for us.
        
        print "The percentage of Cs and Gs for $header, @seq[1] is $percent\n";
  
      }
    }


    }
###Last, but not least, we close our input file and we're done!###
    close FILE

####If all goes well, your output should look something like this:
    The percentage of Cs and Gs for gi|226451773|gb|FJ846591.1| Macaca mulatta isolate ULL5 solute carrier family 25 member A5 (SLC25A5) gene, partial cds,  is 48.0895915678524
    The percentage of Cs and Gs for gi|226451771|gb|FJ846590.1| Macaca mulatta isolate ULL4 solute carrier family 25 member A5 (SLC25A5) gene, partial cds,  is 48.0895915678524
    The percentage of Cs and Gs for gi|226451769|gb|FJ846589.1| Macaca mulatta isolate ULL3 solute carrier family 25 member A5 (SLC25A5) gene, partial cds,  is 48.0895915678524
    The percentage of Cs and Gs for gi|226451767|gb|FJ846588.1| Macaca mulatta isolate ULL2 solute carrier family 25 member A5 (SLC25A5) gene, partial cds,  is 47.9578392621871
    The percentage of Cs and Gs for gi|226451765|gb|FJ846587.1| Macaca mulatta isolate ULL1 solute carrier family 25 member A5 (SLC25A5) gene, partial cds,  is 48.0895915678524
    The percentage of Cs and Gs for gi|226451763|gb|FJ846586.1| Macaca mulatta isolate 425.01 solute carrier family 25 member A5 (SLC25A5) gene, partial cds,  is 48.0895915678524
    The percentage of Cs and Gs for gi|226451761|gb|FJ846585.1| Macaca mulatta isolate 374.03 solute carrier family 25 member A5 (SLC25A5) gene, partial cds,  is 48.0895915678524
    The percentage of Cs and Gs for gi|226451759|gb|FJ846584.1| Macaca mulatta isolate 288.02 solute carrier family 25 member A5 (SLC25A5) gene, partial cds,  is 47.9578392621871
    The percentage of Cs and Gs for gi|226451757|gb|FJ846583.1| Macaca mulatta isolate 228.99 solute carrier family 25 member A5 (SLC25A5) gene, partial cds,  is 48.0895915678524
    The percentage of Cs and Gs for gi|226451755|gb|FJ846582.1| Macaca mulatta isolate 21693 solute carrier family 25 member A5 (SLC25A5) gene, partial cds,  is 48.0895915678524
    The percentage of Cs and Gs for gi|226451753|gb|FJ846581.1| Macaca mulatta isolate 20580 solute carrier family 25 member A5 (SLC25A5) gene, partial cds,  is 48.0895915678524
    The percentage of Cs and Gs for gi|226451751|gb|FJ846580.1| Macaca mulatta isolate 20330 solute carrier family 25 member A5 (SLC25A5) gene, partial cds,  is 48.0895915678524
    The percentage of Cs and Gs for gi|226451749|gb|FJ846579.1| Macaca mulatta isolate 182.92 solute carrier family 25 member A5 (SLC25A5) gene, partial cds,  is 48.0895915678524
    The percentage of Cs and Gs for gi|226451747|gb|FJ846578.1| Macaca mulatta isolate 17414 solute carrier family 25 member A5 (SLC25A5) gene, partial cds,  is 48.0895915678524
    The percentage of Cs and Gs for gi|226451745|gb|FJ846577.1| Macaca fascicularis isolate ULL10 solute carrier family 25 member A5 (SLC25A5) gene, partial cds,  is 47.9578392621871
    The percentage of Cs and Gs for gi|226451743|gb|FJ846576.1| Macaca fascicularis isolate ULL9 solute carrier family 25 member A5 (SLC25A5) gene, partial cds,  is 48.0895915678524
    The percentage of Cs and Gs for gi|226451741|gb|FJ846575.1| Macaca fascicularis isolate ULL8 solute carrier family 25 member A5 (SLC25A5) gene, partial cds,  is 47.2727272727273
    The percentage of Cs and Gs for gi|226451739|gb|FJ846574.1| Macaca fascicularis isolate ULL7 solute carrier family 25 member A5 (SLC25A5) gene, partial cds,  is 47.2727272727273
    The percentage of Cs and Gs for gi|226451737|gb|FJ846573.1| Macaca fascicularis isolate ULL4 solute carrier family 25 member A5 (SLC25A5) gene, partial cds,  is 47.9578392621871
    The percentage of Cs and Gs for gi|226451735|gb|FJ846572.1| Macaca fascicularis isolate ULL3 haplotype 2 solute carrier family 25 member A5 (SLC25A5) gene, partial cds,  is 47.8787878787879


##Thanks for reading! Hope this makes at least *some* sense...
######-Jenn
